optee-teec/macros/src/lib.rs

```diff
@@ -78,50 +111,64 @@ pub fn plugin_invoke(_args: TokenStream, input: TokenStream) -> TokenStream {
     let f = parse_macro_input!(input as syn::ItemFn);
     let f_vis = &f.vis;
     let f_block = &f.block;
-    let f_decl = &f.decl;
-    let f_inputs = &f_decl.inputs;
+    let f_sig = &f.sig;
+    let f_inputs = &f_sig.inputs;
+    let return_type = check_return_type(&f);
 
+    let valid_return_type = return_type == ReturnType::TeecEmptyResult;
```

Comment:
If we only accept the return type as TeecEmptyResult, why we need to define the ReturnType::No and ReturnType::Other?
In line 129 we can just `&& check_return_type()` and 
in check_return_type(): 
syn::ReturnType::Type(xxx) => checks in line92-95; true
_ => false